### DISCUSSION TITLE
Gas fees recommendation/estimation

### DISCUSSION THREAD

**Discussion by malamok-gh** [filecoin-project/lotus#6170] [General Q&A] [answered]
Hi good day,
Sorry I am new here regarding Lotus Filecoin. I got a task that already overdue.

My task has created an API that executes Lotus CLI and returns Gas fees recommendation/estimation, but without inputting any Message information it only depends on network congested/traffic.
He expects general information and recommendation for Gas without using Message information.

Sorry, Does anyone know how can I achieve this or does even posssible? I don't know what should I do, Gas fees computation kind a complicated for me.

API expected Request params/body
N/A

API expected return values
STATUS CODE: 200 OK
{
   "gas": 0.000000005
   "fee_cap": 0.000000005
   "gas_premium": 0.000000005
}

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment, marked as answer]
gas used * baseFee = burn fee
perium * baseFee = miner fee

those two ultimately define the cost of a message. 

you do not know gas used before the message is executed and you do not  know the baseFee either. all you can do is take historic values (gas used per message type for example) and make educated guesses - that the api returns you is the result of a local test run of the message not a final value
